### DISCUSSION TITLE
Detached mode fall back to switch mode?

### DISCUSSION THREAD

**Discussion by HarvsG** [arendst/Tasmota#11662] [Q&A] [answered]
### Question

Is it possible to run a tasmota switch in detached mode (only sends MQTT message, no relay trigger) but if the network is down - or even better, if the MQTT broker is down - it then toggles the relay?

### Reason

I want to have smart LED bulbs that I can automate colour temperature, brightness and on/off state with HomeAssistant, smart speakers etc.

I also *need* to have wife/visitor approval, that means that light-switches need to behave as expected and be useable.

I think that the best way to do this is to replace each light-switch with a tasmota-flashed switch in 'Detached mode'. That way `Wife toggles switch --> Tasmota sends MQTT message to broker --> ZigBee message to smart bulb --> smart bulb toggled`

_However_ what if the WiFi or broker goes down? Now I want the switch to actually trigger the relay and break the circuit, else I'm left unable to change the state of the light. Now I want `Wife toggles switch --> Tasmota sends MQTT message --> no response --> Tasmota toggles relay --> light changes state(--> happy wife --> happy life)`.

I think I have read this is possible but I would have no Idea where to start.

### Links
- [Reditors think this is possible with ESPhome](https://www.reddit.com/r/homeautomation/comments/jddyue/comment/g97vvxk) with [ESPhome example config](https://gist.github.com/DelusionalAI/c3bfa05b5268a96f8a0c527ceace5f1b#file-gistfile1-txt)
-  [So do HA users](https://community.home-assistant.io/t/make-esphome-node-fallback-when-not-connected-to-ha-api/116615)
- Looks as if a `ping` based [watchdog rule](https://tasmota.github.io/docs/Rules/#watchdog-for-wi-fi-router) could be used but not sure how to configure and no guarantee the MQTT broker is up

**Comment by barbudor** [comment, marked as answer]
Depending how you detached
Have you read this : https://tasmota.github.io/docs/Buttons-and-Switches/#switchtopic
The warning seems to address your question.

Otherwise you could use rules with `on mqttconnected ...` and `on mqtt#disconnected ...` to adapt to your liking

**Reply by HarvsG** [reply]
Yes I did read that page, having not yet used Tasmota I didn't quite understand it. Is it saying the behaviour I want is the default behaviour?

>When using SwitchTopic 1 or 2 (or ButtonTopic 1 or 2) and your MQTT broker becomes unavailable, Tasmota falls back to default SwitchTopic 0 (or ButtonTopic 0), which is not optimal.
To avoid this, we recommend using first two options instead

I'm not sure what options is is referring to when it says 'first two options'

**Reply by barbudor** [reply]
The default behavior for a button or switch, when not detached is to directly drive the matching relay. Which I understand is what you are looking for.

The "options" it refers are the 3 options/methods to detach the switch/buttons from the relay :
1- Setoption73/Setoption114
2- Rules
3- Switchtopic/buttontopic

Given you **want** the switch/button to directly drive the relay when the MQTT/Wifi is not reachable, the "warning" should be understood as you should use the 3rd option.

Many things will become more and more clear when getting your hands dirty with Tasmota, MQTT and all

**Reply by HarvsG** [reply]
Many thanks for your time, I will mark this as answered. Now I just need to find a switch which supports my needs: Tasmota, no neutral wire and supports low-power LED lights.

---
 > The default behavior for a button or switch, when not detached is to directly drive the matching relay. Which I understand is what you are looking for.

This is only the behaviour I'm looking for when MQTT is unreachable

**Reply by barbudor** [reply]
It can be also useful to have a Nodemcu around to experiment

**Reply by HarvsG** [reply]
>Given you want the switch/button to directly drive the relay when the MQTT/Wifi is not reachable, the "warning" should be understood as you should use the 3rd option.

So, in summary, `SwitchMode 0`, `SwitchTopic 1` Will send an MQTT `TOGGLE` on every state change but do nothing to the relay. In the event of MQTT being down `SwitchTopic` will default to 0 and toggle the relay with every state change, giving exactly the behaviour I want?

Presumably, when the MQTT comes back online `SwitchTopic 1` behaviour is restored?

**Reply by barbudor** [reply]
Just to be sure we are aligned:
`SwitchMode0` is meant to be used with a rocker switch so each time the switch will be flipped it will generate a TOOGLE. It's not meant to be used with push-button.
You may need this setting but it's not related to the "detached" subject.

`SwitchTopic 1` (or `SwitchTopic customtopic`) is exactly as your describe:
- when MQTT is available => only publish, no relay control
- when MQTT is down => relay control
